CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+HTTPS:`git clone https://github.com/redox-os/redox.git`<br>
+SSH:`git clone git@github.com:redox-os/redox.git --origin upstream --recursive`<br>
+Use HTTPS if you don't know which one to use. (Recommended: learn about SSH if you don't want to have to login every time you push/pull!)<br>
+3. Add your fork with<br>
+HTTPS:`git remote add origin https://github.com/your-username/redox.git`<br>
+SSH:`git remote add origin git@github.com:your-username/redox.git --origin upstream --recursive`<br>
+4. Optionally create a separate branch (recommended if you're making multiple changes simultaneously)<br>
+5. Make changes<br>
+6. Commit (`git add . --all; git commit -m "my commit"`)<br>
+7. Test your changes with `make qemu` or `make virtualbox` (you might have to use `make qemu_no_kvm`)<br>
+8. Pull from upstream (`git fetch upstream; git rebase upstream/master`) (Note: try not to use `git pull`, it is equivalent to doing `git fetch upstream; git merge`, which is not usually preferred for local repositories.)<br>
+9. Repeat step 7 to make sure the rebase still works<br>
+10. Push to your fork (`git push origin my-branch`)<br>
+11. Create a pull request<br><br>
+12. Describe your changes<br>
+13. Submit!<br>
+
+<a name="best-practices"/>
+## Best Practices/Guidelines
+<!-- TODO add this section to the index/TOC -->
+
+#### General
+<!-- TODO fill out this section -->
+
+#### Testing Practices
+
+- It's always better to test boot every time you make a change, because it is important to see how the OS boots and works after it compiles. Even though Rust is a safety-oriented language, something as unstable as an in-dev operating system will have problems in many cases and may completely break on even the slightest critical change. Also, make sure you check how the unmodified version runs on your machine before making any changes. Else, you won't have anything to compare to, and it will generally just lead to confusion. TLDR; Build/rebuild often.
+
+- To run the ZFS tests:
+    - Create the zfs.img only once. If one has not been created, run `make filesystem/apps/zfs/zfs.img` before booting into Redox.
+    - Run `open zfs.img` to open the created ZFS image.
+    - Run `file /home/LICENSE.md` twice to ensure ARC isn't broken.
+
+#### Kernel
+
+- When trying to access a slice, **always** use the `common::GetSlice` trait and use the `.get_slice()` method to get a slice without causing the kernel to panic. The problem with slicing in regular Rust, e.g. `foo[a..b]`, is that if someone tries to access with a range that is out of bounds of an array/string/slice, it will cause a panic at runtime, as a safety measure. Same thing when accessing an element. Always use `foo.get(n)` instead of `foo[n]` and try to cover for the possibility of `Option::None`. Doing the regular way may work fine for applications, but never in the kernel. No possible panics should ever exist in kernel space.
```

Comment:
Well, the line written here is specifically meant for the kernel, hence why I point to `common::GetSlice` and explain the cases. As a general rule somewhere else in the file, @polymetric1 could specify that you could use `get_slice`/`get`/etc in user space to prevent panicking as an optional thing to do.